File: swarmmanagercontroller/src/main/java/co/uk/swarmbit/docker/api/networks/NetworksApiImpl.java
package co.uk.swarmbit.docker.api.networks;

import co.uk.swarmbit.docker.api.common.AbstractApiImpl;
import co.uk.swarmbit.docker.api.common.json.NetworkJson;
import co.uk.swarmbit.docker.api.networks.parameters.NetworkListParameters;
import co.uk.swarmbit.docker.api.common.json.NetworkCreateResponseJson;
import co.uk.swarmbit.docker.api.networks.parameters.NetworkCreateParameters;
import co.uk.swarmbit.rest.RestResponseType;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class NetworksApiImpl extends AbstractApiImpl implements NetworksApi {

    private static final String NETWORKS_PATH = "/networks";

    private static final String CREATE_PATH = "/create";

    @Override
    public List<NetworkJson> listNetworks(String swarmId, NetworkListParameters parameters) {
        return listObjects(NETWORKS_PATH, swarmId, new RestResponseType<List<NetworkJson>>(){}, parameters);
    }

    @Override
    public NetworkJson inspectNetwork(String swarmId, String id) {
        return inspectObject(NETWORKS_PATH + "/" + id, swarmId, new RestResponseType<NetworkJson>(){});
    }

    @Override
    public NetworkCreateResponseJson createNetwork(String swarmId, NetworkCreateParameters parameters) {
        return createObject(NETWORKS_PATH + CREATE_PATH, swarmId, new RestResponseType<NetworkCreateResponseJson>(){}, parameters);
    }

    @Override
    public void deleteNetwork(String swarmId, String id) {
        deleteObject(NETWORKS_PATH + "/" + id, swarmId, new RestResponseType<Void>(){});
    }
}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/networks/NetworksApiImpl.kt
package com.swarmbit.docker.api.networks

import com.swarmbit.docker.api.common.AbstractApiImpl
import com.swarmbit.docker.api.common.client.DockerWebClient
import com.swarmbit.docker.api.common.rest.model.ResponseType
import com.swarmbit.docker.api.networks.model.Network
import com.swarmbit.docker.api.networks.model.NetworkCreateResponse
import com.swarmbit.docker.api.networks.parameters.NetworksCreateParameters
import com.swarmbit.docker.api.networks.parameters.NetworksInspectParameters
import com.swarmbit.docker.api.networks.parameters.NetworksListParameters
import io.micronaut.context.annotation.Context

@Context
class NetworksApiImpl(dockerWebClient: DockerWebClient) : AbstractApiImpl(dockerWebClient), NetworksApi {
    companion object {
        private const val NETWORKS_PATH = "/networks"
        private const val CREATE_PATH = "/create"
    }

    override fun listNetworks(swarmId: String, parameters: NetworksListParameters): List<Network> {
        return listObjects(NETWORKS_PATH, swarmId, object : ResponseType<List<Network>>() {}, parameters)
    }

    override fun inspectNetwork(swarmId: String, id: String, parameters: NetworksInspectParameters): Network {
        return inspectObject(NETWORKS_PATH, swarmId, object : ResponseType<Network>() {}, id, parameters)
            ?: throw IllegalArgumentException("No network found for swarmId ($swarmId) and networkId (${id})")
    }

    override fun createNetwork(swarmId: String, parameters: NetworksCreateParameters): NetworkCreateResponse {
        return createObject(NETWORKS_PATH + CREATE_PATH, swarmId, object : ResponseType<NetworkCreateResponse>() {}, parameters)!!
    }

    override fun deleteNetwork(swarmId: String, id: String) {
        deleteObject("$NETWORKS_PATH/$id", swarmId, object : ResponseType<Void>() {})
    }

}
